File: ctci/ch1/ex1_is_unique/is_unique.py

# Returns true if a string has all unique characters
def is_unique(string):
    charDict = dict()
    for char in string:
        if char in charDict:
            return False
        charDict[char] = 1
    return True

if __name__== "__main__":
    if is_unique("tesrgu"):
        print('Is Unique')
    else:
        print('Is Not Unique')

File: ctci/ch1/ex1_is_unique/is_unique.c
#include <stdio.h>

char isUnique(char *s) {
    int charArray[26];
    for (int i=0; i < 26; ++i) {
        charArray[i] = 0;
    }
    char c = *s++;
    while(c) {
        if (charArray['z' - c]) {
            return 0;
        }
        charArray['z' - c] = 1;
        c = *s++;
    }
    return 1;
}

int main() {
    char result = isUnique("tesi");
    if (result) {
        printf("%s\n", "Is Unique");
    } else {
        printf("%s\n", "Is Not Unique");
    }
    return 0;
}